### PR TITLE
Add test to verify multicall rate limit scenarios

### PR DESCRIPTION
**Pull request by dchymko**
## Description

Adding a test to validate the behaviour in #1155. Will probably want to wait for it to be merged before reviewing this

### CODE REVIEW

**Comment by rinatkhaziev** [comment]
@dchymko think we waited enough? :)

**Comment by rinatkhaziev** [comment]
@dchymko 

Seems it fails on `$result = $myxmlrpcserver->multiCall( $method_calls );` 

I have 0 context, so I'm not sure how to proceed. 

If you have a quick minute to fix the test, I'd appreciate that, otherwise, we can mark this as `won't merge` and close.

```
There was 1 error:

1) VIP_Go_Security_Test::test__login_system_multicall_rate_limit
Cannot modify header information - headers already sent by (output started at /tmp/wordpress-tests-lib/includes/bootstrap.php:122)

/app/security.php:125
/tmp/wordpress/wp-includes/class-wp-hook.php:289
/tmp/wordpress/wp-includes/plugin.php:206
/tmp/wordpress/wp-includes/class-wp-xmlrpc-server.php:292
/tmp/wordpress/wp-includes/class-wp-xmlrpc-server.php:4747
/tmp/wordpress/wp-includes/class-wp-xmlrpc-server.php:663
/tmp/wordpress/wp-includes/IXR/class-IXR-server.php:109
/tmp/wordpress/wp-includes/IXR/class-IXR-server.php:206
/app/tests/test-security.php:146

ERRORS!
Tests: 8, Assertions: 15, Errors: 1.
```

**Comment by dchymko** [comment]
> @dchymko
> 
> Seems it fails on `$result = $myxmlrpcserver->multiCall( $method_calls );`
> 
> I have 0 context, so I'm not sure how to proceed.
> 
> If you have a quick minute to fix the test, I'd appreciate that, otherwise, we can mark this as `won't merge` and close.
> 
> ```
> There was 1 error:
> 
> 1) VIP_Go_Security_Test::test__login_system_multicall_rate_limit
> Cannot modify header information - headers already sent by (output started at /tmp/wordpress-tests-lib/includes/bootstrap.php:122)

Huh, it looks like something might have changed in how this works once it was merged.  What I could do is mimic what happens in [WP_XMLRPC_UnitTestCase](https://core.trac.wordpress.org/browser/trunk/tests/phpunit/tests/xmlrpc/basic.php) but I'm not sure if that's worth the effort.  I'll try it out and see if I can get it to work but if it's going to be a lot effort we might just want to close this out.

File: tests/test-security.php [hunk 1]
```diff
@@ -116,6 +119,40 @@ public function test__lost_password_limit() {
...
+		$limit_threshold = 50;
+		$last_threshold_index = $limit_threshold - 1;
+
+		for ( $i = 1; $i <= $limit_threshold + 2; $i++ ) {
+			array_push( $method_calls, $method );
+		}
+
+		$myxmlrpcserver->callbacks = $myxmlrpcserver->methods;
+
+		$result = $myxmlrpcserver->multiCall( $method_calls );
```
Comment (dchymko) [inline tests/test-security.php:146]:
```suggestion
		$this->_error_level = error_reporting();
		error_reporting( $this->_error_level & ~E_WARNING );
		$result = $myxmlrpcserver->multiCall( $method_calls );
		error_reporting( $this->_error_level );
```
Suppress the `headers already sent` error for this test.

**Comment by rinatkhaziev** [comment]
Thanks for doing another pass at this, @dchymko, per our conversation I'm marking this as "wontfix" and closing.

**Pull request closed without merging**